Code/Yospace/Source/YospaceManager.swift

```diff
@@ -0,0 +1,170 @@
+//
+//  YospaceManager.swift
+//
+//  Created by Raffi on 29/05/2024.
+//  Copyright © 2024 THEOplayer. All rights reserved.
+//
+
+import THEOplayerSDK
+import YOAdManagement
+
+class YospaceManager {
+    let player: THEOplayerSDK.THEOplayer
+    let eventDispatcher: EventDispatcher = .init()
+    var didSetSourceFromConnector: Bool = false
+    private(set) var adIntegrationController: THEOplayerSDK.ServerSideAdIntegrationController?
+    private var adIntegrationHandler: THEOplayerSDK.ServerSideAdIntegrationHandler?
+    private var yospaceSession: YOSession?
+    private var source: YospaceManagerSource?
+    private var id3MetadataHandler: YospaceID3MetadataHandler?
+    private var playerEventsHandler: THEOplayerEventsHandler?
+    private var yospaceNotificationsHandler: YospaceNotificationsHandler?
+    private(set) var isSettingSource: Bool = false
+
+    private typealias YospaceManagerSource = (THEOplayerSDK.SourceDescription, THEOplayerSDK.TypedSource)
+
+    init(player: THEOplayerSDK.THEOplayer) {
+		self.player = player
+
+        self.player.ads.registerServerSideIntegration(integrationId: "yospace") { controller in
```

Comment:
should not this be the other way around?
so customers should just call something like
`player.ads.registerServerSideIntegration(integrationId: "yospace", YospaceConnector())`?

and I think we should also make sure that they can pass the [YospaceSessionProperties](https://developer.yospace.com/sdk-documentation/apple/userguide/latest/en/reference-materials.html#sdk-yosessionproperties)

Reply:
Yeah, for me it felt a bit more logical that we _promote_ our new API, so we ask the users to explicitly register their SSAI integration on the player (instead of being _hidden_ inside the "YospaceConnector"). Then probably we also don't need to ask them to hold a reference to the connector instance anymore, as the player will do it.

But if it was like this already, or this is the same on the other platforms, fine by me.

We can add the `YOSessionProperties` to the `init`, indeed.

Reply:
Done

Reply:
On Android, [I added `sessionProperties` to `YospaceSsaiDescription`](https://github.com/THEOplayer/android-connector/blob/7.6.1/connectors/yospace/src/main/java/com/theoplayer/android/connector/yospace/YospaceSsaiDescription.kt#L30), so you can set it in your `SourceDescription` ([see second example](https://www.theoplayer.com/docs/theoplayer/connectors/android/yospace/getting-started/#usage)). That way, you can have different session properties for different Yospace sources.

Could we do something similar on iOS?

Reply:
> should not this be the other way around?
> so customers should just call something like
> `player.ads.registerServerSideIntegration(integrationId: "yospace", YospaceConnector())`?

My reasoning was that if we add another integration API in the future, then the `YospaceConnector` constructor could register itself on *multiple* APIs. If we let the customer call `registerServerSideIntegration`, then we couldn't do that and we'd have to ask our customers to update their code instead.

Reply:
I have moved the YoSessionProperties to YospaceServerSideAdInsertionConfiguration.